pyro/distributions/util.py

```diff
@@ -188,16 +188,22 @@ def torch_sign(value):
     return torch.sign(value)
 
 
-def log_sum_exp(tensor, dim=-1):
-    """
-    Numerically stable implementation for the `LogSumExp` operation. The
-    summing is done along the dimension specified by ``dim``.
-
-    :param torch.Tensor tensor: Input tensor.
-    :param dim: Dimension to be summed out.
-    """
-    max_val = tensor.max(dim)[0]
-    return max_val + (tensor - max_val.unsqueeze(dim)).exp().sum(dim=dim).log()
+try:
+    from torch import logsumexp  # for pytorch 0.4.1 and later
+except ImportError:
+    def logsumexp(tensor, dim=-1):
+        """
+        Numerically stable implementation for the `LogSumExp` operation. The
+        summing is done along the dimension specified by ``dim``.
+
+        :param torch.Tensor tensor: Input tensor.
+        :param dim: Dimension to be summed out.
+        """
+        max_val = tensor.max(dim)[0]
+        return max_val + (tensor - max_val.unsqueeze(dim)).exp().sum(dim=dim).log()
+
+
+log_sum_exp = logsumexp  # DEPRECATED
```

Comment:
Since this isn't part of the public API, I suppose we can take the liberty of deleting it with the update?

Reply:
We're actually using this internally in a customer pipeline, and I'd like to update that pipeline before deleting this alias.

Reply:
Got it, makes sense.